wayback-core/src/main/java/org/archive/wayback/authenticationcontrol/IPMatchesBooleanOperator.java

```diff
@@ -62,15 +65,69 @@ public void setAllowedRanges(List<String> allowedRanges) {
 			}
 		}
 	}
+	
+	/**		
+	 * @return null. this is a placeholder for Spring's getter/setter 		
+	 * 			examination		
+	 */
+	public List<String> getTrustedProxies() {
+		return null;
+	}
+
+	/**
+	 * @param trustedProxies parses each String IPRange provided for the proxies, adding them to
+	 * 		the list of IPRanges which must be ignored by the IP match operator
+	 */
+	public void setTrustedProxies(List<String> trustedProxies) {
+		this.trustedProxies = new ArrayList<IPRange>();
+		for (String ip : trustedProxies) {
+			IPRange range = new IPRange();
+			if (range.setRange(ip)) {
+				this.trustedProxies.add(range);
+			} else {
+				LOGGER.severe("Unable to parse range (" + ip + ")");
+			}
+		}
+	}
+
+
+	public String getClientIPFromForwardedForHeader(String forwardedForHeader){
+		ArrayList<String> forwardingIPs;
+		String ip = null;
+		Boolean containsIP = false;
+		if (forwardedForHeader.contains(",")) {
+			forwardingIPs = new ArrayList<String>(Arrays.asList(forwardedForHeader.replace(" ", "").split(",")));
+			Collections.reverse(forwardingIPs);
+			for (String forwardingIP : forwardingIPs){
+				for (IPRange range : trustedProxies) {
+					if (range.contains(forwardingIP)) {
+						containsIP = true;
+						break;
+					}
+				}
+				if (!containsIP ||
+					(containsIP && forwardingIPs.get(forwardingIPs.size() -1).equals(forwardingIP))) {
```

Comment:
Trivial, but we should add whitespace after `-` operator in `forwardingIPs.size() -1`.